Unsubscribe Link produces error when mail body not yet loaded

<!-- WARNING: All issues created by non-contributors WILL be closed.
  -- Please use https://community.getmailspring.com/ for all bug reports
  -- and feature requests.
  -->

## Objective
When opening an old e-mail whose mail body is not loaded (not in the local cache?), the unsubcsribe link that is added to the e-mail view crashes with the following error.

This should be resolved before the next release, as this release newly introduces the unsubscribe link.

```
usr/share/mailspring/resources/app.asar/node_modules/react-dom/cjs/react-dom.production.min.js:181 Error: cheerio.load() expects a string
    at Function.exports.load (/usr/share/mailspring/resources/app.asar/node_modules/cheerio/lib/static.js:30)
    at bestUnsubscribeLink (/tmp/nylas-build/electron-packager/linux-x64/mailspring-linux-x64/resources/app/internal_packages/list-unsubscribe/lib/main.tsx:69)
    at UnsubscribeContainer (/tmp/nylas-build/electron-packager/linux-x64/mailspring-linux-x64/resources/app/internal_packages/list-unsubscribe/lib/main.tsx:93)
    at eh (/usr/share/mailspring/resources/app.asar/node_modules/react-dom/cjs/react-dom.production.min.js:169)
    at ai (/usr/share/mailspring/resources/app.asar/node_modules/react-dom/cjs/react-dom.production.min.js:202)
    at bi (/usr/share/mailspring/resources/app.asar/node_modules/react-dom/cjs/react-dom.production.min.js:203)
    at Fi (/usr/share/mailspring/resources/app.asar/node_modules/react-dom/cjs/react-dom.production.min.js:218)
    at Gi (/usr/share/mailspring/resources/app.asar/node_modules/react-dom/cjs/react-dom.production.min.js:217)
    at ii (/usr/share/mailspring/resources/app.asar/node_modules/react-dom/cjs/react-dom.production.min.js:214)
    at xg (/usr/share/mailspring/resources/app.asar/node_modules/react-dom/cjs/react-dom.production.min.js:212)
    at Object.enqueueSetState (/usr/share/mailspring/resources/app.asar/node_modules/react-dom/cjs/react-dom.production.min.js:133)
    at MessageList.Component.setState (/usr/share/mailspring/resources/app.asar/node_modules/react/cjs/react.development.js:373)
    at _MessageStore.MessageList._onChange (/tmp/nylas-build/electron-packager/linux-x64/mailspring-linux-x64/resources/app/internal_packages/message-list/lib/message-list.tsx:330)
    at EventEmitter.eventHandler (/tmp/nylas-build/electron-packager/linux-x64/mailspring-linux-x64/resources/app/src/global/mailspring-store.ts:119)
    at EventEmitter.emit (events.js:215)
    at EventEmitter.emit (domain.js:476)
    at _MessageStore.trigger (/tmp/nylas-build/electron-packager/linux-x64/mailspring-linux-x64/resources/app/src/global/mailspring-store.ts:132)
    at /tmp/nylas-build/electron-packager/linux-x64/mailspring-linux-x64/resources/app/src/flux/stores/message-store.ts:309
Eh @ /usr/share/mailspring/resources/app.asar/node_modules/react-dom/cjs/react-dom.production.min.js:181
/usr/share/mailspring/resources/app.asar/src/error-logger.js:101 Error: cheerio.load() expects a string
    at Function.exports.load (/usr/share/mailspring/resources/app.asar/node_modules/cheerio/lib/static.js:30)
    at bestUnsubscribeLink (/tmp/nylas-build/electron-packager/linux-x64/mailspring-linux-x64/resources/app/internal_packages/list-unsubscribe/lib/main.tsx:69)
    at UnsubscribeContainer (/tmp/nylas-build/electron-packager/linux-x64/mailspring-linux-x64/resources/app/internal_packages/list-unsubscribe/lib/main.tsx:93)
    at eh (/usr/share/mailspring/resources/app.asar/node_modules/react-dom/cjs/react-dom.production.min.js:169)
    at ai (/usr/share/mailspring/resources/app.asar/node_modules/react-dom/cjs/react-dom.production.min.js:202)
    at bi (/usr/share/mailspring/resources/app.asar/node_modules/react-dom/cjs/react-dom.production.min.js:203)
    at Fi (/usr/share/mailspring/resources/app.asar/node_modules/react-dom/cjs/react-dom.production.min.js:218)
    at Gi (/usr/share/mailspring/resources/app.asar/node_modules/react-dom/cjs/react-dom.production.min.js:217)
    at ii (/usr/share/mailspring/resources/app.asar/node_modules/react-dom/cjs/react-dom.production.min.js:214)
    at xg (/usr/share/mailspring/resources/app.asar/node_modules/react-dom/cjs/react-dom.production.min.js:212)
    at Object.enqueueSetState (/usr/share/mailspring/resources/app.asar/node_modules/react-dom/cjs/react-dom.production.min.js:133)
    at MessageList.Component.setState (/usr/share/mailspring/resources/app.asar/node_modules/react/cjs/react.development.js:373)
    at _MessageStore.MessageList._onChange (/tmp/nylas-build/electron-packager/linux-x64/mailspring-linux-x64/resources/app/internal_packages/message-list/lib/message-list.tsx:330)
    at EventEmitter.eventHandler (/tmp/nylas-build/electron-packager/linux-x64/mailspring-linux-x64/resources/app/src/global/mailspring-store.ts:119)
    at EventEmitter.emit (events.js:215)
    at EventEmitter.emit (domain.js:476)
    at _MessageStore.trigger (/tmp/nylas-build/electron-packager/linux-x64/mailspring-linux-x64/resources/app/src/global/mailspring-store.ts:132)
    at /tmp/nylas-build/electron-packager/linux-x64/mailspring-linux-x64/resources/app/src/flux/stores/message-store.ts:309 Object
```

## Testing Notes
Ensure that an old e-mail whose body is not loaded is used for testing.

## Tips
<!-- Any insights that can assist with this task? -->

## References
Bug was introduced in this PR: https://github.com/Foundry376/Mailspring/pull/2275

--- a/app/internal_packages/list-unsubscribe/lib/main.tsx
+++ b/app/internal_packages/list-unsubscribe/lib/main.tsx
@@ -66,24 +66,30 @@ interface UnsubscribeAction {
 }
 
 function bestUnsubscribeLink(message): string {
-  const dom = cheerio.load(message.body);
-  const links = _getLinks(dom);
 
   let result = null;
 
-  for (const link of links) {
-    for (const re of regexps) {
-      if (re.test(link.href)) {
-        // If the URL contains e.g. "unsubscribe" we assume that we have correctly
-        // detected the unsubscribe link.
-        return link.href;
-      }
-      if (re.test(link.innerText)) {
-        // If the URL does not contain "unsubscribe", but the text around the link contains
-        // it, it is a possible candidate, we we still check all other links for a better match
-        result = link.href;
+  // Only check the body if it has been downloaded already
+  if (message.body) {
+
+    const dom = cheerio.load(message.body);
+    const links = _getLinks(dom);
+
+    for (const link of links) {
+      for (const re of regexps) {
+        if (re.test(link.href)) {
+          // If the URL contains e.g. "unsubscribe" we assume that we have correctly
+          // detected the unsubscribe link.
+          return link.href;
+        }
+        if (re.test(link.innerText)) {
+          // If the URL does not contain "unsubscribe", but the text around the link contains
+          // it, it is a possible candidate, we we still check all other links for a better match
+          result = link.href;
+        }
       }
     }
+
   }
 
   return result;

